### PR TITLE
Docs: Fix white flash on iframe load for Safari

### PR DESCRIPTION
**Pull request by JeremyJaydan**
**Description**
Navigating the documentation while using Safari the iframe responsible for loading page content flashes white as it's loading (see attached screen recoding).

https://github.com/user-attachments/assets/3a3570dd-9fef-43f1-89d2-77a418040f80

Safari Version 17.6 (19618.3.11.11.5)

**Solution**
Move the iframe "display none" out of the if condition so it's hidden before the src is set and "display unset" on iframe load/error regardless of the if condition. Not sure if the error listener is really needed but kept it there to be safe.

Edit (https://github.com/mrdoob/three.js/pull/29419/commits/6834f41f074359f9979e64979d88e95488984bbd):
Moved the load/error event listeners to inside the if condition, which keeps the iframe hidden when there's no page selected.

https://github.com/user-attachments/assets/50dd18e4-3b49-42fc-b992-c97bb97d7eee

I'm new to contributing, let me know if there's anything else to do!

### CODE REVIEW

**Comment by JeremyJaydan** [comment]
Re: https://github.com/mrdoob/three.js/pull/29419/commits/6834f41f074359f9979e64979d88e95488984bbd

Before:
Screenshot 2024-09-16 at 2 08 38 PM

After:
Screenshot 2024-09-16 at 2 09 14 PM

This was an oversight introduced by this PR which is fixed by the mentioned commit.

File: docs/index.html [hunk 1]
```diff
@@ -529,17 +529,29 @@ <h1><a href="https://threejs.org">three.js</a></h1>
...
+
 			if ( hash && titles[ splitHash[ 0 ] ] ) {
 
+				iframe.onload = function () {
+
+					iframe.style.display = 'unset';
+
+				};
+
+				iframe.onerror = function () {
```
Comment (Mugen87) [inline docs/index.html:542]: I can confirm the white flash in Safari and that the PR fixes the issue!

Just one question: Why do we need an `onerror` event listener? What error situation triggers the callback?

Comment (JeremyJaydan) [reply docs/index.html:542]: I was thinking with the `onerror` would cover cases such as 404 or network failure and show default browser pages for the iframe to show but I did test both those cases and they didn't seem to work.

Perhaps the `onerror` should be removed as there's no proven use case for it so far.

Comment (Mugen87) [reply docs/index.html:542]: > Perhaps the onerror should be removed as there's no proven use case for it so far.

Sounds good!

Comment (JeremyJaydan) [reply docs/index.html]: done :smile:

**Review by Mugen87 — APPROVED** [review]

**Pull request merged**